### PR TITLE
Use Hikari rather than Commons DBCP in DataSource configuration examples

### PR DESCRIPTION
**Pull request by rajadilipkolli**
From spring boot 2.x, the default DataSource library that comes with starter is Hikari, documentation is using commons-dbcp. To avoid confusion updating documentation to use `HikariDataSource`

### CODE REVIEW

**Comment by philwebb** [comment]
Thanks!

**Comment by snicoll** [comment]
@rajadilipkolli thanks for the PR but I am wondering what you see as confusing? This documentation you've updated is about configuring two datasources. These are manual configurations, and as such, don't really care about what the default in auto-configuration would be. 

The documented examples also use two different datasources to show that's something possible if needed. I don't really see an interest at moving everything to Hikari.

**Comment by rajadilipkolli** [comment]
Hi @snicoll , when I copied the examples from documentation. It didn't work as import is not there then I realized that from starter only Hikari is there. 

so, either we should change the example or update the documentation that we can use any choice of data source.

**Comment by snicoll** [comment]
> . It didn't work as import is not there then I realized that from starter only Hikari is there.

Yes. This is an advanced configuration example of configuring multiple datasources. I don't believe that the starter changes from one datasource to the other is the problem here.

You can't just copy/paste an example from the ref doc in "any" app and expect it to work. We could document that you need to add the relevant driver jar for the example to work but I personally am not sure that we need to go into that level of detail for an advanced example like this. 

Let's see what the rest of the team thinks.

**Comment by mhalbritter** [comment]
I agree with @snicoll. This in an advanced example which shows the power of this approach, so that you can even configure two totally independent datasources (in this case Hikari and Commons DBCP).

We could maybe add a HINT: that explains that this example configures two unrelated connection pools?

**Comment by philwebb** [comment]
I agree with @wilkinsona, I think consistently using Hikari would be best.

File: spring-boot-project/spring-boot-docs/src/main/java/org/springframework/boot/docs/howto/dataaccess/configuretwodatasources/MyAdditionalDataSourceConfiguration.java [hunk 1]
```diff
@@ -31,7 +31,7 @@ public class MyAdditionalDataSourceConfiguration {
 	@Bean(defaultCandidate = false)
 	@ConfigurationProperties("app.datasource")
 	public BasicDataSource secondDataSource() {
```
[the remark this answers is not in the document]
Comment (rajadilipkolli) [reply spring-boot-project/spring-boot-docs/src/main/java/org/springframework/boot/docs/howto/dataaccess/configuretwodatasources/MyAdditionalDataSourceConfiguration.java]: Hi @wilkinsona , I have updated PR to `HikariDataSource` but shouldn't it be `DataSource` ? 

I feel `DataSource` is more accurate here.

**Comment by wilkinsona** [comment]
Thanks for your efforts here, @rajadilipkolli. Unfortunately, things seem to be getting further away from what we want. For example, the latest commits make some changes to some of the tests that shouldn't not be necessary. They also unnecessarily change a configuration property prefix which means that the names no longer align with those in the documented example.

I'm afraid we don't have time to guide you through the necessary changes step-by-step. To receive that level of guidance, please keep an eye out for an issue that's labelled as ideal-for-contribution. I'm going to close this one now but thanks anyway for your efforts here and bringing the potential for improvement to our attention. I've opened https://github.com/spring-projects/spring-boot/issues/43054 to tackle it.

**Pull request closed without merging**